src/main/java/org/commcare/formplayer/services/ScheduledTasks.java

```diff
@@ -0,0 +1,48 @@
+package org.commcare.formplayer.services;
+
+import com.timgroup.statsd.StatsDClient;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.formplayer.Application;
+import org.commcare.formplayer.repo.FormSessionRepo;
+import org.commcare.formplayer.util.Constants;
+import org.commcare.formplayer.util.RequestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
+
+@Component
+public class ScheduledTasks {
+
+    private final Log log = LogFactory.getLog(ScheduledTasks.class);
+
+    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss");
+
+    @Autowired
+    protected FormSessionRepo formSessionRepo;
+
+    @Autowired
+    private StatsDClient datadogStatsDClient;
+
+    // the default "-" corresponds to Scheduled.CRON_DISABLED
+    @Scheduled(cron= "${commcare.formplayer.scheduledTasks.purge.cron:-}")
+    @SchedulerLock(name = "purge",
+            lockAtMostFor = "${commcare.formplayer.scheduledTasks.purge.lockAtMostFor:5h}",
+            lockAtLeastFor = "${commcare.formplayer.scheduledTasks.purge.lockAtLeastFor:20m}")
+    public void purge() {
+        log.info("Starting purge scheduled task.");
+        int deletedRows = formSessionRepo.purgeFormSessions();
+        datadogStatsDClient.count(
+                "%s.%s".format(Constants.SCHEDULED_TASKS_PURGE, "deletedRows"),
+                deletedRows
+        );
+        datadogStatsDClient.increment(
```

Comment:
Danny would have a better idea than myself about the datadog reporting event structure, but the bucketing of tracking this way seems slightly odd to me, since the number of times run will effectively be a constant. It does seem useful to know if the process isn't succeeding, but I don't think this would actually capture that since failures are effectively swallowed in the `purgeFormSessions` call

Reply:
This was mostly to ensure that things are run at most once daily. With the `deletedRows` counter, we might have more than 1 tasks scheduled/reporting at the same time and we wouldn't know how many it came from.

Reply:
Ah! That makes sense as a metric that we want to be keeping track of, thanks for clarifying.